framework/src/meshgenerators/BlockDeletionGenerator.C

```diff
@@ -20,20 +19,50 @@ BlockDeletionGenerator::validParams()
 {
   InputParameters params = ElementDeletionGeneratorBase::validParams();
 
-  params.addClassDescription(
-      "Mesh generator which removes elements with the specified subdomain ID");
-  params.addRequiredParam<SubdomainID>("block_id", "The block to be deleted");
+  params.addClassDescription("Mesh generator which removes elements from the specified subdomains");
+  params.addParam<std::vector<SubdomainName>>("block", "The list of blocks to be deleted");
+  params.addDeprecatedParam<SubdomainID>(
+      "block_id", "The block to be deleted.", "Use block instead");
 
   return params;
 }
 
 BlockDeletionGenerator::BlockDeletionGenerator(const InputParameters & parameters)
-  : ElementDeletionGeneratorBase(parameters), _block_id(getParam<SubdomainID>("block_id"))
+  : ElementDeletionGeneratorBase(parameters)
+{
+  // Handle deprecated parameter
+  if (isParamValid("block_id"))
+    _block_ids.push_back(getParam<SubdomainID>("block_id"));
+  if (!isParamValid("block_id") && !isParamValid("block"))
+    mooseError("Must provide the blocks to be deleted in the 'block' parameter");
+  if (isParamValid("block_id") && isParamValid("block"))
+    paramError("block_id", "Cannot use with the parameter 'block'. Please use just 'block'.");
+}
+
+std::unique_ptr<MeshBase>
+BlockDeletionGenerator::generate()
 {
+  if (isParamValid("block"))
+    // Get the list of block ids from the block names
+    _block_ids =
+        MooseMeshUtils::getSubdomainIDs(*_input, getParam<std::vector<SubdomainName>>("block"));
+
+  // Check that the block ids/names exist in the mesh
+  std::set<SubdomainID> mesh_blocks;
+  _input->subdomain_ids(mesh_blocks);
+
+  for (std::size_t i = 0; i < _block_ids.size(); ++i)
+    if (_block_ids[i] == Moose::INVALID_BLOCK_ID || !mesh_blocks.count(_block_ids[i]))
+      paramError("block",
+                 "The block '",
+                 getParam<std::vector<SubdomainName>>("block")[i],
+                 "' was not found within the mesh");
```

Comment:
You have to wrap this all in brackets.

The error checking will not work if `block_id` is provided, because the parameter `block` doesn't exist.